serde/src/ser/impls.rs

```diff
@@ -98,6 +100,28 @@ impl Serialize for String {
 
 ///////////////////////////////////////////////////////////////////////////////
 
+#[cfg(feature = "std")]
+impl Serialize for CStr {
+    #[inline]
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        (self.to_bytes_with_nul()).serialize(serializer)
+    }
+}
+
+#[cfg(feature = "std")]
+impl Serialize for CString {
+    #[inline]
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where S: Serializer
+    {
+        (self.to_bytes_with_nul()).serialize(serializer)
```

Comment:
Using [`serialize_bytes`](https://docs.serde.rs/serde/ser/trait.Serializer.html#tymethod.serialize_bytes) would allow most serializers to serialize this more efficiently than other types of slices.